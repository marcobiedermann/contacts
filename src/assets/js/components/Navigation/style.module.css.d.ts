declare const styles: {
  readonly navigation: string;
  readonly navigation__list: string;
  readonly 'navigation__list--inline': string;
};
export = styles;
