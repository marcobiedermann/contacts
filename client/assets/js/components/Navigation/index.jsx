import React from 'react';
import { useTranslation } from 'react-i18next';
import { NavLink } from 'react-router-dom';
import * as routes from '../../constants/routes';

const Navigation = () => {
  const { t } = useTranslation();

  return (
    <nav>
      <ul>
        <li>
          <NavLink
            to={routes.INDEX}
            activeClassName="is-active"
            exact
          >
            {t('pages:home')}
          </NavLink>
        </li>
        <li>
          <NavLink
            to={routes.CONTACTS}
            activeClassName="is-active"
            exact
          >
            {t('pages:contacts')}
          </NavLink>
        </li>
        <li>
          <NavLink
            to={routes.NEW}
            activeClassName="is-active"
            exact
          >
            {t('pages:new')}
          </NavLink>
        </li>
        <li>
          <NavLink
            to={routes.LOGIN}
            activeClassName="is-active"
            exact
          >
            {t('pages:login')}
          </NavLink>
        </li>
        <li>
          <NavLink
            to={routes.REGISTER}
            activeClassName="is-active"
            exact
          >
            {t('pages:register')}
          </NavLink>
        </li>
      </ul>
    </nav>
  );
};

export default Navigation;
