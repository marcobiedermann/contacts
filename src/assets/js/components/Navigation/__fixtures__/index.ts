import { NavigationProps } from '..';

const props: NavigationProps = {
  routes: [
    {
      name: 'Home',
      path: '/',
    },
  ],
};

export default props;
