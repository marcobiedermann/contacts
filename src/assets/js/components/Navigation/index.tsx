import classNames from 'classnames';
import React, { FC } from 'react';
import { useTranslation } from 'react-i18next';
import { NavLink } from 'react-router-dom';
import { CONTACTS, CONTACT_NEW, LOGIN, LOGOUT, REGISTER } from '../../constants/routes';
import styles from './style.module.css';

export interface NavigationProps {
  inline?: boolean;
}

const Navigation: FC<NavigationProps> = (props) => {
  const { inline } = props;
  const { t } = useTranslation();

  return (
    <nav className={styles.navigation}>
      <ul
        className={classNames(styles.navigation__list, {
          [styles['navigation__list--inline']]: inline,
        })}
      >
        <li>
          <NavLink to={CONTACTS.path} activeClassName="is-active" exact>
            {t('pages:contacts')}
          </NavLink>
        </li>
        <li>
          <NavLink to={CONTACT_NEW.path} activeClassName="is-active" exact>
            {t('pages:new')}
          </NavLink>
        </li>
        <li>
          <NavLink to={LOGIN.path} activeClassName="is-active" exact>
            {t('pages:login')}
          </NavLink>
        </li>
        <li>
          <NavLink to={REGISTER.path} activeClassName="is-active" exact>
            {t('pages:register')}
          </NavLink>
        </li>
        <li>
          <NavLink to={LOGOUT.path} activeClassName="is-active" exact>
            {t('pages:logout')}
          </NavLink>
        </li>
      </ul>
    </nav>
  );
};

export default Navigation;
