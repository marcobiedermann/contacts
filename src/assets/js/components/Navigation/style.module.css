.navigation {
}

.navigation__list {
  gap: 1em;
  list-style: none;
  margin: 0;
  padding: 0;
}

.navigation__list--inline {
  display: flex;
}
