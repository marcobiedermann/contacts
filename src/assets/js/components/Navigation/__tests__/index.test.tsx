import { render } from '@testing-library/react';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import Navigation from '..';
import props from '../__fixtures__';

describe('navigation component', () => {
  it('should render correctly', () => {
    expect.assertions(1);

    const { container } = render(
      <MemoryRouter>
        <Navigation {...props} />
      </MemoryRouter>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
