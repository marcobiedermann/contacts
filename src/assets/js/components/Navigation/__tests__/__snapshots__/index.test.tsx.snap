// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navigation component should render correctly 1`] = `
<DocumentFragment>
  <nav
    class="navigation"
  >
    <ul
      class="navigation__list"
    >
      <li>
        <a
          aria-current="page"
          class="is-active"
          href="/"
        >
          Home
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
